ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/admin/UserManagementClient.scala

```diff
@@ -0,0 +1,111 @@
+      Some(UserManagementClient.toProtoUser(user)),
+      initialRights.view.map(UserManagementClient.toApiRight).toList
+    )
+    LedgerClient
+      .stub(service, token)
+      .createUser(request)
+      .map(UserManagementClient.fromProtoUser)
+  }
+
+  def getUser(userId: UserId, token: Option[String] = None): Future[User] =
+    LedgerClient
+      .stub(service, token)
+      .getUser(GetUserRequest(userId.toString))
+      .map(UserManagementClient.fromProtoUser)
+
+  /** Retrieve the User information for the user authenticated by the token(s) on the call . */
+  def getAuthenticatedUser(token: Option[String] = None): Future[User] =
+    LedgerClient
+      .stub(service, token)
+      .getUser(GetUserRequest())
+      .map(UserManagementClient.fromProtoUser)
+
+  def deleteUser(userId: UserId, token: Option[String] = None): Future[Unit] =
+    LedgerClient
+      .stub(service, token)
+      .deleteUser(proto.DeleteUserRequest(userId.toString))
+      .map(_ => ())
+
+  def listUsers(token: Option[String] = None): Future[Vector[User]] =
+    LedgerClient
+      .stub(service, token)
+      .listUsers(proto.ListUsersRequest())
+      .map(_.users.view.map(fromProtoUser).toVector)
+
+  def grantUserRights(userId: UserId, rights: Seq[UserRight], token: Option[String] = None): Future[Vector[UserRight]] =
+    LedgerClient
+      .stub(service, token)
+      .grantUserRights(proto.GrantUserRightsRequest(userId.toString, rights.map(toApiRight)))
+      .map(_.newlyGrantedRights.view.map(fromApiRight).toVector)
+
+  def revokeUserRights(userId: UserId, rights: Seq[UserRight], token: Option[String] = None): Future[Vector[UserRight]] =
+    LedgerClient
+      .stub(service, token)
+      .revokeUserRights(proto.RevokeUserRightsRequest(userId.toString, rights.map(toApiRight)))
+      .map(_.newlyRevokedRights.view.map(fromApiRight).toVector)
+
+  def listUserRights(userId: UserId, token: Option[String] = None): Future[Vector[UserRight]] =
+    LedgerClient
+      .stub(service, token)
+      .listUserRights(proto.ListUserRightsRequest(userId.toString))
+      .map(_.rights.view.map(fromApiRight).toVector)
+
+  /** Retrieve the rights of the user authenticated by the token(s) on the call . */
+  def listAuthenticatedUserRights(token: Option[String] = None): Future[Vector[UserRight]] =
+    LedgerClient
+      .stub(service, token)
+      .listUserRights(proto.ListUserRightsRequest())
+      .map(_.rights.view.map(fromApiRight).toVector)
+}
+
+object UserManagementClient {
+  private def fromProtoUser(user: proto.User): User =
+    User(
+      UserId(Ref.UserId.assertFromString(user.id)),
+      Option.unless(user.primaryParty.isEmpty)(Party.assertFromString(user.primaryParty))
+    )
+
+  private def toProtoUser(user: User): proto.User =
+    proto.User(user.id.toString, user.primaryParty.fold("")(_.toString))
+
+  // FIXME: move to the right place in the codebase; and cleanup imports; and dedup with the code in 'ApiUserManagementService'
```

Comment:
I’m fine with addressing that after merging. At least within the language team we’ve found it useful to include a link to an issue for that type of stuff so you can easily grep for it later.

Reply:
I've fixed this now.